lib/LoginStrategy.ts

```diff
@@ -43,14 +43,14 @@ export class LoginStrategy extends BaseStrategy {
   private readonly jwtSecret?: string | Buffer
   private readonly jwtSignOptions?: jwt.SignOptions
   private readonly provider: providers.JsonRpcProvider
-  private readonly cacheServerClient!: CacheServerClient
   private readonly numberOfBlocksBack: number
   private readonly ensResolver: PublicResolver
   private readonly didResolver: Resolver
   private readonly ipfsStore: DidStore
   private readonly acceptedRoles: Set<string>
   private readonly strategyAddress?: string
   private readonly privateKey: string
+  private cacheServerClient?: CacheServerClient
```

Comment:
@Harasz So is this the key part of the change? Before we had `cacheServerClient` as an asserted not-undefined and not it can be undefined?

Reply:
The key part is that we are set a reference to `cacheServerClient` after login to cache server not before. That change was just to make types more predictable.